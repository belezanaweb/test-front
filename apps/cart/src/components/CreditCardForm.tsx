import { Controller, useForm } from 'react-hook-form';
import { useNavigate } from 'react-router-dom';

import { yupResolver } from '@hookform/resolvers/yup';

import { useCartProvider } from '../contexts/CartContext';
import { CreditCard } from '../models/CreditCard';
import { Button } from './Button';
import { CartInformation } from './CartInformation';
import { Input } from './Input';
import { InputMasked } from './InputMasked';
import { Layout } from './Layout';
import { schema } from './schema';

export type FormFields = CreditCard;

export function CreditCardForm() {
  const navigate = useNavigate();
  const { cart, setCreditCard } = useCartProvider();

  const {
    register,
    handleSubmit,
    control,
    formState: { errors, isValid },
  } = useForm<FormFields>({
    mode: 'all',
    reValidateMode: 'onBlur',
    resolver: yupResolver(schema),
  });

  const onSubmit = handleSubmit(async ({ number, holder, expireDate, cvv }) => {
    setCreditCard({ number, holder, expireDate, cvv });
    navigate('/information');
  });

  return (
    <>
      <form
        className="flex flex-col bg-white rounded-lg border border-stone-300 p-4 w-full"
        onSubmit={onSubmit}
        id="credit-card-form"
      >
        <h1 className="text-black text-xl mb-4">Cartão de crédito</h1>

        <Controller
          name="number"
          control={control}
          render={({ fieldState, field }) => (
            <InputMasked
              id="number"
              label="Número"
              placeholder="0000 0000 0000 0000"
              className="mb-5"
              format="#### #### #### ####"
              value={field.value}
              onChange={field.onChange}
              error={fieldState.error?.message}
            />
          )}
        />

        <Input
          id="holder"
          label="Nome do titular do cartão"
          placeholder="Nome impresso no cartão"
          className="mb-5"
          {...register('holder')}
          error={errors.holder?.message}
        />

        <Controller
          name="expireDate"
          control={control}
          render={({ fieldState, field }) => (
            <InputMasked
              id="expireDate"
              label="Data de validade"
              placeholder="MM/AA"
              className="mb-5"
              format="##/##"
              mask={['M', 'M', 'Y', 'Y']}
              value={field.value}
              onChange={field.onChange}
              error={fieldState.error?.message}
            />
          )}
        />

        <Controller
          name="cvv"
          control={control}
          render={({ fieldState, field }) => (
            <InputMasked
              id="cvv"
              label="Código CVV:"
              placeholder="000"
              format="###"
              value={field.value}
              onChange={field.onChange}
              error={fieldState.error?.message}
            />
          )}
        />
      </form>

      <Layout.Footer>
        <CartInformation
          productsCount={cart?.items.length ?? 0}
          subtotal={cart?.subTotal ?? 0}
          shipping={cart?.shippingTotal ?? 0}
          discount={cart?.discount ?? 0}
          total={cart?.total ?? 0}
        >
          <Button type="submit" form="credit-card-form" disabled={!isValid}>
            Finalizar pedido
          </Button>
        </CartInformation>
      </Layout.Footer>
    </>
  );
}
