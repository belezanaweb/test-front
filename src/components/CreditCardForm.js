import React from 'react';

import Style from '../styles/Root';

function CreditCardForm(){

  return(
    <h1>Credit Card Form</h1>
  )
}

export default CreditCardForm;