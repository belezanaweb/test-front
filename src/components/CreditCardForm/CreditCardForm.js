import React from 'react'
import Input from '../Input/Input'
import { createUseStyles } from 'react-jss'

const useStyles = createUseStyles({
  creditCardForm: {
    width: '100%',
    padding: '12px 10px 0',
    margin: '0 auto',
    boxSizing: 'border-box',
    display: 'flex',
    '& form': {
      width: '100%'
    }
  },
  doubleColumn: {
    display: 'flex',
    flexWrap: 'wrap',
    '& div:first-child': {
      width: '55%',
      marginRight: '5%'
    },
    '& div:last-child': {
      width: '40%'
    }
  }
})

const CreditCardForm = (props) => {
  const classes = useStyles()
  const { disableSubmit } = props

  return (
    <div className={classes.creditCardForm} data-testid="CreditCardForm">
      <form>
        <Input labelText="Número do Cartão" type="number" role="creditCard" />

        <Input labelText="Nome do Titular" type="text" role="holderName" />

        <div className={classes.doubleColumn}>
          <Input labelText="CVV" type="number" role="cvv" />
          <Input labelText="Validade (mês/ano)" type="text" role="expiryDate" />
        </div>
      </form>
    </div>
  )
}

export default CreditCardForm
