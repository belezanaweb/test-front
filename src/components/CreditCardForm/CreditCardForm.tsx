import React, { useContext } from 'react';
import { Controller } from 'react-hook-form';
import styled from 'styled-components';

import Box, { Grid } from '../Box';
import TextField from '../TextField';

import { CreditCardFormContext } from './CreditCardFormController';

const StyledForm = styled.div`
  flex: 1;
`;

const CreditCardForm = () => {
  const context = useContext(CreditCardFormContext);

  if (context === null) return null;

  const { control, errors } = context;

  function renderFieldNumber() {
    const mask = [/\d/, /\d/, /\d/, /\d/, '.', /\d/, /\d/, /\d/, /\d/, '.', /\d/, /\d/, /\d/, /\d/, '.', /\d/, /\d/, /\d/, /\d/];
    return (
      <Controller
        as={
          <TextField
            label="Número do Cartão:"
            mask={mask}
            error={errors.number && errors.number.message}
            placeholder="____.____.____.____"
          />
        }
        name="number"
        rules={{ required: "Numero do cartão obrigatório." }}
        control={control}
      />
    );
  }

  function renderFieldName() {
    return (
      <Controller
        as={
          <TextField
            label="Nome do Titular:"
            mask={false}
            error={errors.name && errors.name.message}
            placeholder="Como no cartão"
          />
        }
        name="name"
        rules={{ required: "Nome do titular obrigatório." }}
        control={control}
      />
    );
  }

  function renderFieldExpiringDate() {
    return (
      <Controller
        as={
          <TextField
            label="Validade (mês/ano):"
            mask={[/\d/, /\d/, '/', /\d/, /\d/, /\d/, /\d/]}
            error={errors.expiringDate && errors.expiringDate.message}
            placeholder="__/____"
          />
        }
        name="expiringDate"
        rules={{ required: "Validade do cartão obrigatório." }}
        control={control}
      />
    );
  }

  function renderFieldCodeVerification() {
    return (
      <Controller
        as={
          <TextField
            error={errors.code && errors.code.message}
            label="CVV:"
            mask={[/\d/, /\d/, /\d/]}
            placeholder="___"
          />
        }
        name="code"
        rules={{ required: "Codígo de verificação obrigatório." }}
        control={control}
      />
    );
  }

  return (
    <StyledForm>
      <Grid template="1fr" gap="25px" css={{ marginBottom: '25px' }}>
        <Box>
          {renderFieldNumber()}
        </Box>

        <Box>
          {renderFieldName()}
        </Box>
      </Grid>

      <Grid template="1.2fr 1fr" css={{ marginBottom: '15px' }}>
        <Box>
          {renderFieldExpiringDate()}
        </Box>
        <Box>
          {renderFieldCodeVerification()}
        </Box>
      </Grid>
    </StyledForm>
  );
};

export default CreditCardForm;
