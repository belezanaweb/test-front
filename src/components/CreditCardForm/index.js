import React, { useState, useCallback } from 'react'

import { useCreditCard } from '../../hooks/creditCard'

import Card from '../Card'
import Input from '../Input'

import { Form, Warper } from './styles'

const CreditCardForm = ({ formId, formSubmitted }) => {
  const { formData } = useCreditCard()

  const [error, setError] = useState({})

  const handleSubmit = (e) => {
    e.preventDefault()
    if (FormDataIsValid()) formSubmitted()
  }

  const FormDataIsValid = useCallback(() => {
    let errorObj = { valid: false }

    Object.keys(formData).forEach((data) => {
      let isValid

      switch (data) {
        case 'number':
          isValid = formData.number
            ?.replaceAll('.', '')
            .match(
              /^(?:(4[0-9]{12}(?:[0-9]{3})?)|(5[1-5][0-9]{14})|(6(?:011|5[0-9]{2})[0-9]{12})|(3[47][0-9]{13})|(3(?:0[0-5]|[68][0-9])[0-9]{11})|((?:2131|1800|35[0-9]{3})[0-9]{11}))$/g
            )
          break
        case 'name':
          isValid = formData.name?.match(/^[a-zA-Z]{2,30}$/g)
          break
        case 'expiry':
          isValid = formData.expiry?.match(/[0-9/]{7}/g)
          break
        case 'cvv':
          isValid = formData.cvv?.match(/\d{3}/g)
          break
        default:
          break
      }
      errorObj = { ...errorObj, [data]: !isValid, valid: isValid }
    })

    setError(errorObj)

    return errorObj.valid
  }, [formData])

  return (
    <Card>
      <Form id={formId} onSubmit={handleSubmit}>
        <Input
          name={'number'}
          onlyNumbers
          placeholder={'____.____.____.____'}
          maskPattern={'####.####.####.####'}
          maskDivider={'.'}
          label={'Número do cartão:'}
          hasError={error.number}
        />
        <Input
          name={'name'}
          placeholder={'Como no cartão'}
          label={'Nome do Titular:'}
          hasError={error.name}
        />
        <Warper>
          <Input
            name={'expiry'}
            onlyNumbers
            placeholder={'__/____'}
            maskPattern={'##/####'}
            maskDivider={'/'}
            label={'Validade (mês/ano):'}
            hasError={error.expiry}
          />
          <Input
            name={'cvv'}
            onlyNumbers
            placeholder={'___'}
            maskPattern={'###'}
            label={'CVV:'}
            hasError={error.cvv}
          />
        </Warper>
      </Form>
    </Card>
  )
}

export default CreditCardForm
