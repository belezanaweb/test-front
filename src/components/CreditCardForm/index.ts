export { default } from './CreditCardForm';
export { default as CreditCardFormController, CreditCardFormContext } from './CreditCardFormController';
