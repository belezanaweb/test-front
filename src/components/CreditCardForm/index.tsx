import React from 'react'

import { useCreditCard } from '../../hooks/creditCard'

import Card from '../Card'
import Input from '../Input'

import { Form, Warper } from './styles'

interface Props {
  formId: string
  formSubmitted: () => void
}

const CreditCardForm: React.FC<Props> = ({ formId, formSubmitted }) => {

  const { error } = useCreditCard()

  const handleSubmit = (e: React.FormEvent<HTMLFormElement> | undefined) => {
    e?.preventDefault()
    formSubmitted()
  }

  return (
    <Card>
      <Form id={formId} onSubmit={handleSubmit}>
        <Input
          name={'number'}
          onlyNumbers
          placeholder={'____.____.____.____'}
          maskPattern={'####.####.####.####'}
          maskDivider={'.'}
          label={'Número do cartão:'}
          errorLabel={'Número do cartão inválido'}
          hasError={error.number}
        />
        <Input
          name={'name'}
          placeholder={'Como no cartão'}
          label={'Nome do Titular:'}
          errorLabel={'Nome de titular inválido'}
          hasError={error.name}
        />
        <Warper>
          <Input
            name={'expiry'}
            onlyNumbers
            placeholder={'__/____'}
            maskPattern={'##/####'}
            maskDivider={'/'}
            label={'Validade (mês/ano):'}
            errorLabel={'Data inválida'}
            hasError={error.expiry}
          />
          <Input
            name={'cvv'}
            onlyNumbers
            placeholder={'___'}
            maskPattern={'###'}
            label={'CVV:'}
            errorLabel={'CVV inválido'}
            hasError={error.cvv}
          />
        </Warper>
      </Form>
    </Card>
  )
}

export default CreditCardForm
