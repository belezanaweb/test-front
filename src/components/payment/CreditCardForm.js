import React, { Component } from 'react'
import { Link } from 'react-router-dom'

export default class CreditCardForm extends Component {
  state = {
    creditCard: '',
    name: '',
    validate: '',
    cvv: ''
  }

  onChange = e => {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  render() {
    const { creditCard, name, validate, cvv } = this.state
    return (
      <div>
        <form onSubmit={this.onSubmit}>
          <div className="form-group">
            <label>Numero do cartao</label>
            <input
              className="form-control"
              type="number"
              name="creditCard"
              onChange={this.onChange}
              value={creditCard}
            />
          </div>
          <div className="form-group">
            <label>Nome</label>
            <input
              className="form-control"
              type="text"
              name="name"
              onChange={this.onChange}
              value={name}
            />
          </div>
          <div className="form-group">
            <label>Validade</label>
            <input
              className="form-control"
              type="number"
              name="validate"
              onChange={this.onChange}
              value={validate}
            />
            <label>CVV</label>
            <input
              className="form-control"
              type="number"
              name="cvv"
              onChange={this.onChange}
              value={cvv}
            />
          </div>
          {creditCard !== '' ? (
            name !== '' ? (
              validate !== '' ? (
                cvv !== '' ? (
                  <Link to="/request">FINALIZAR PEDIDO</Link>
                ) : (
                  ''
                )
              ) : (
                ''
              )
            ) : (
              ''
            )
          ) : (
            ''
          )}
        </form>
      </div>
    )
  }
}
