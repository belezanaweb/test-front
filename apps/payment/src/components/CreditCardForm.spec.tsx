import { fireEvent, render, screen, waitFor } from '@testing-library/react';

import CreditCardForm from './CreditCardForm';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useNavigate: () => mockedUsedNavigate,
}));

describe('CreditCardForm', () => {
  it('should render the form fields correctly', () => {
    const { container } = render(<CreditCardForm />);

    expect(screen.getByLabelText('Número')).toBeInTheDocument();
    expect(
      screen.getByLabelText('Nome do titular do cartão'),
    ).toBeInTheDocument();
    expect(screen.getByLabelText('Data de validade')).toBeInTheDocument();
    expect(screen.getByLabelText('Código CVV:')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should form initialize with empty values, empty errors and 'disabled' submit button", () => {
    render(<CreditCardForm />);

    expect(screen.getByLabelText('Número')).toHaveValue('');
    expect(screen.getByLabelText('Nome do titular do cartão')).toHaveValue('');
    expect(screen.getByLabelText('Data de validade')).toHaveValue('');
    expect(screen.getByLabelText('Código CVV:')).toHaveValue('');

    expect(screen.queryAllByText(/span/i).length).toBe(0);

    expect(
      screen.getByRole('button', { name: /Finalizar pedido/i }),
    ).toBeDisabled();
  });

  it('should form validate the fields correctly', async () => {
    const { container } = render(<CreditCardForm />);

    fireEvent.change(screen.getByLabelText('Número'), {
      target: { value: '1234567890123456' },
    });
    fireEvent.change(screen.getByLabelText('Nome do titular do cartão'), {
      target: { value: 'John' },
    });
    fireEvent.change(screen.getByLabelText('Data de validade'), {
      target: { value: '12/24' },
    });
    fireEvent.change(screen.getByLabelText('Código CVV:'), {
      target: { value: '13' },
    });

    await waitFor(() => {
      expect(
        screen.getByText('O número do cartão é inválido.'),
      ).toBeInTheDocument();
      expect(
        screen.getByText('O nome do titular é inválido.'),
      ).toBeInTheDocument();
      expect(
        screen.getByText('A data de validade é inválida.'),
      ).toBeInTheDocument();
      expect(
        screen.getByText('O código de segurança é inválido.'),
      ).toBeInTheDocument();
    });

    expect(
      screen.getByRole('button', { name: /Finalizar pedido/i }),
    ).toBeDisabled();
    fireEvent.click(screen.getByRole('button', { name: /Finalizar pedido/i }));
    expect(mockedUsedNavigate).not.toHaveBeenCalled();
    expect(container.firstChild).toMatchSnapshot();

    fireEvent.change(screen.getByLabelText('Número'), {
      target: { value: '4444333322221111' },
    });
    fireEvent.change(screen.getByLabelText('Nome do titular do cartão'), {
      target: { value: 'John Doe' },
    });
    fireEvent.change(screen.getByLabelText('Data de validade'), {
      target: { value: '12/2024' },
    });
    fireEvent.change(screen.getByLabelText('Código CVV:'), {
      target: { value: '123' },
    });

    await waitFor(() => {
      expect(
        screen.queryByText('O número do cartão é inválido.'),
      ).not.toBeInTheDocument();
      expect(
        screen.queryByText('O nome do titular é inválido.'),
      ).not.toBeInTheDocument();
      expect(
        screen.queryByText('A data de validade é inválida.'),
      ).not.toBeInTheDocument();
      expect(
        screen.queryByText('O código de segurança é inválido.'),
      ).not.toBeInTheDocument();
    });

    expect(
      screen.getByRole('button', { name: /Finalizar pedido/i }),
    ).not.toBeDisabled();

    fireEvent.click(screen.getByRole('button', { name: /Finalizar pedido/i }));

    await waitFor(() => {
      expect(container.firstChild).toMatchSnapshot();
      expect(mockedUsedNavigate).toHaveBeenCalledWith('/information');
      expect(mockedUsedNavigate).toHaveBeenCalledTimes(1);
    });
  });
});
