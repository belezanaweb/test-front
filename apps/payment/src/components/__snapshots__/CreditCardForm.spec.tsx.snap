// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditCardForm should form validate the fields correctly 1`] = `
<form
  class="flex flex-col bg-white rounded-lg border border-stone-300 p-4 w-full"
  id="credit-card-form"
>
  <h1
    class="text-black text-xl mb-4"
  >
    Cartão de crédito
  </h1>
  <div
    class="flex flex-col mb-5"
  >
    <label
      class="text-neutral-600 text-sm mb-1"
      for="number"
    >
      Número
    </label>
    <input
      class="w-full h-11 rounded shadow-inner border px-4 text-black text-sm leading-none outline-none border-red-600"
      id="number"
      inputmode="numeric"
      placeholder="0000 0000 0000 0000"
      type="text"
      value="1234 5678 9012 3456"
    />
    <span
      class="text-red-600 text-xs leading-none mt-1"
    >
      O número do cartão é inválido.
    </span>
  </div>
  <div
    class="flex flex-col mb-5"
  >
    <label
      class="text-neutral-600 text-sm mb-1"
      for="holder"
    >
      Nome do titular do cartão
    </label>
    <input
      class="w-full h-11 rounded shadow-inner border px-4 text-black text-sm leading-none outline-none border-red-600"
      id="holder"
      name="holder"
      placeholder="Nome impresso no cartão"
    />
    <span
      class="text-red-600 text-xs leading-none mt-1"
    >
      O nome do titular é inválido.
    </span>
  </div>
  <div
    class="flex flex-1 mb-5 gap-4"
  >
    <div
      class="flex flex-col w-full"
    >
      <label
        class="text-neutral-600 text-sm mb-1"
        for="expireDate"
      >
        Data de validade
      </label>
      <input
        class="w-full h-11 rounded shadow-inner border px-4 text-black text-sm leading-none outline-none border-red-600"
        id="expireDate"
        inputmode="numeric"
        placeholder="MM/AAAA"
        type="text"
        value="12/24YY"
      />
      <span
        class="text-red-600 text-xs leading-none mt-1"
      >
        A data de validade é inválida.
      </span>
    </div>
    <div
      class="flex flex-col w-full"
    >
      <label
        class="text-neutral-600 text-sm mb-1"
        for="cvv"
      >
        Código CVV:
      </label>
      <input
        class="w-full h-11 rounded shadow-inner border px-4 text-black text-sm leading-none outline-none border-red-600"
        id="cvv"
        inputmode="numeric"
        placeholder="000"
        type="text"
        value="13 "
      />
      <span
        class="text-red-600 text-xs leading-none mt-1"
      >
        O código de segurança é inválido.
      </span>
    </div>
  </div>
</form>
`;

exports[`CreditCardForm should form validate the fields correctly 2`] = `
<form
  class="flex flex-col bg-white rounded-lg border border-stone-300 p-4 w-full"
  id="credit-card-form"
>
  <h1
    class="text-black text-xl mb-4"
  >
    Cartão de crédito
  </h1>
  <div
    class="flex flex-col mb-5"
  >
    <label
      class="text-neutral-600 text-sm mb-1"
      for="number"
    >
      Número
    </label>
    <input
      class="w-full h-11 rounded shadow-inner border border-stone-300 px-4 text-black text-sm leading-none outline-none"
      id="number"
      inputmode="numeric"
      placeholder="0000 0000 0000 0000"
      type="text"
      value="4444 3333 2222 1111"
    />
  </div>
  <div
    class="flex flex-col mb-5"
  >
    <label
      class="text-neutral-600 text-sm mb-1"
      for="holder"
    >
      Nome do titular do cartão
    </label>
    <input
      class="w-full h-11 rounded shadow-inner border border-stone-300 px-4 text-black text-sm leading-none outline-none"
      id="holder"
      name="holder"
      placeholder="Nome impresso no cartão"
    />
  </div>
  <div
    class="flex flex-1 mb-5 gap-4"
  >
    <div
      class="flex flex-col w-full"
    >
      <label
        class="text-neutral-600 text-sm mb-1"
        for="expireDate"
      >
        Data de validade
      </label>
      <input
        class="w-full h-11 rounded shadow-inner border border-stone-300 px-4 text-black text-sm leading-none outline-none"
        id="expireDate"
        inputmode="numeric"
        placeholder="MM/AAAA"
        type="text"
        value="12/2024"
      />
    </div>
    <div
      class="flex flex-col w-full"
    >
      <label
        class="text-neutral-600 text-sm mb-1"
        for="cvv"
      >
        Código CVV:
      </label>
      <input
        class="w-full h-11 rounded shadow-inner border border-stone-300 px-4 text-black text-sm leading-none outline-none"
        id="cvv"
        inputmode="numeric"
        placeholder="000"
        type="text"
        value="123"
      />
    </div>
  </div>
</form>
`;

exports[`CreditCardForm should form validate the fields correctly 3`] = `
<form
  class="flex flex-col bg-white rounded-lg border border-stone-300 p-4 w-full"
  id="credit-card-form"
>
  <h1
    class="text-black text-xl mb-4"
  >
    Cartão de crédito
  </h1>
  <div
    class="flex flex-col mb-5"
  >
    <label
      class="text-neutral-600 text-sm mb-1"
      for="number"
    >
      Número
    </label>
    <input
      class="w-full h-11 rounded shadow-inner border border-stone-300 px-4 text-black text-sm leading-none outline-none"
      id="number"
      inputmode="numeric"
      placeholder="0000 0000 0000 0000"
      type="text"
      value="4444 3333 2222 1111"
    />
  </div>
  <div
    class="flex flex-col mb-5"
  >
    <label
      class="text-neutral-600 text-sm mb-1"
      for="holder"
    >
      Nome do titular do cartão
    </label>
    <input
      class="w-full h-11 rounded shadow-inner border border-stone-300 px-4 text-black text-sm leading-none outline-none"
      id="holder"
      name="holder"
      placeholder="Nome impresso no cartão"
    />
  </div>
  <div
    class="flex flex-1 mb-5 gap-4"
  >
    <div
      class="flex flex-col w-full"
    >
      <label
        class="text-neutral-600 text-sm mb-1"
        for="expireDate"
      >
        Data de validade
      </label>
      <input
        class="w-full h-11 rounded shadow-inner border border-stone-300 px-4 text-black text-sm leading-none outline-none"
        id="expireDate"
        inputmode="numeric"
        placeholder="MM/AAAA"
        type="text"
        value="12/2024"
      />
    </div>
    <div
      class="flex flex-col w-full"
    >
      <label
        class="text-neutral-600 text-sm mb-1"
        for="cvv"
      >
        Código CVV:
      </label>
      <input
        class="w-full h-11 rounded shadow-inner border border-stone-300 px-4 text-black text-sm leading-none outline-none"
        id="cvv"
        inputmode="numeric"
        placeholder="000"
        type="text"
        value="123"
      />
    </div>
  </div>
</form>
`;

exports[`CreditCardForm should render the form fields correctly 1`] = `
<form
  class="flex flex-col bg-white rounded-lg border border-stone-300 p-4 w-full"
  id="credit-card-form"
>
  <h1
    class="text-black text-xl mb-4"
  >
    Cartão de crédito
  </h1>
  <div
    class="flex flex-col mb-5"
  >
    <label
      class="text-neutral-600 text-sm mb-1"
      for="number"
    >
      Número
    </label>
    <input
      class="w-full h-11 rounded shadow-inner border border-stone-300 px-4 text-black text-sm leading-none outline-none"
      id="number"
      inputmode="numeric"
      placeholder="0000 0000 0000 0000"
      type="text"
      value=""
    />
  </div>
  <div
    class="flex flex-col mb-5"
  >
    <label
      class="text-neutral-600 text-sm mb-1"
      for="holder"
    >
      Nome do titular do cartão
    </label>
    <input
      class="w-full h-11 rounded shadow-inner border border-stone-300 px-4 text-black text-sm leading-none outline-none"
      id="holder"
      name="holder"
      placeholder="Nome impresso no cartão"
    />
  </div>
  <div
    class="flex flex-1 mb-5 gap-4"
  >
    <div
      class="flex flex-col w-full"
    >
      <label
        class="text-neutral-600 text-sm mb-1"
        for="expireDate"
      >
        Data de validade
      </label>
      <input
        class="w-full h-11 rounded shadow-inner border border-stone-300 px-4 text-black text-sm leading-none outline-none"
        id="expireDate"
        inputmode="numeric"
        placeholder="MM/AAAA"
        type="text"
        value=""
      />
    </div>
    <div
      class="flex flex-col w-full"
    >
      <label
        class="text-neutral-600 text-sm mb-1"
        for="cvv"
      >
        Código CVV:
      </label>
      <input
        class="w-full h-11 rounded shadow-inner border border-stone-300 px-4 text-black text-sm leading-none outline-none"
        id="cvv"
        inputmode="numeric"
        placeholder="000"
        type="text"
        value=""
      />
    </div>
  </div>
</form>
`;
